src/libs/actions/MapboxTokens.js

```diff
@@ -0,0 +1,89 @@
+import _ from 'underscore';
+import moment from 'moment';
+import Onyx from 'react-native-onyx';
+import {AppState} from 'react-native';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import CONST from '../../CONST';
+
+let connectionID;
+let currentToken;
+let refreshTimeoutID;
+const refreshInterval = 1000 * 60 * 25;
+
+const refreshToken = () => {
+    console.debug('[MapboxTokens] refreshing token every 25 minutes', refreshInterval);
+
+    // Cancel any previous timeouts so that there is only one request to get a token at a time.
+    clearTimeout(refreshTimeoutID);
+
+    // Refresh the token every 25 minutes
+    refreshTimeoutID = setTimeout(() => {
+        console.debug('[MapboxTokens] Fetching a new token after waiting 25 minutes');
```

Comment:
One thing that I realized, which I forgot to plan for, is that this could still run when signed out. Right?

So I think we should return early if the user is not signed in. Otherwise we'll keep refreshing once every 25 minutes and set errors on the token object.

Reply:
That's a good point! Good catch.

Reply:
Yep I just tested and it keeps refreshing while signed out. Let's add that as a test step.